### PR TITLE
Issue 3287: Refactoring the test_script_result table to use react js.

### PR DESCRIPTION
**Pull request by czack810150**

### CODE REVIEW

**Comment by czack810150** [comment]
As per discussion with David on Slack.
(Code is not completed yet)

File: app/javascript/packs/application.js [hunk 1]
```diff
@@ -22,9 +22,11 @@ import { makeExamScanLogTable } from 'javascripts/Components/exam_scan_log_table
 import { makeMarksSpreadsheet } from 'javascripts/Components/marks_spreadsheet';
 import { makeSubmissionFileManager } from 'javascripts/Components/submission_file_manager';
 import { makeRepoBrowser } from 'javascripts/Components/repo_browser';
+import { makeTestScriptResultTable } from 'javascripts/Components/test_script_result_table';
 import { makeSubmissionTable } from 'javascripts/Components/submission_table';
 import { makeMarksGradersManager } from 'javascripts/Components/marks_graders_manager';
 
+
```
Comment (david-yz-liu) [inline app/javascript/packs/application.js]: Delete this, and in general don't change whitespace.

File: app/views/automated_tests/_test_runs.html.erb [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+    document.addEventListener('DOMContentLoaded', () => {
+      makeTestScriptResultTable(document.getElementById('test_script_result_table'));
+    })
+  </script>
+<% end %>
+
+
 <h2><%= t('automated_tests.test_results') %></h2>
+
+
```
Comment (david-yz-liu) [inline app/views/automated_tests/_test_runs.html.erb]: See my above comment. In general, you should stop adding blank lines as a habit.

File: app/assets/javascripts/Components/test_script_result_table.jsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        <h2>beeppeppe</h2>
+        <ReactTable
+          data={data}
+          columns={[
+            {
+              Header: "Runs",
+              accessor: "test_run_id"
+            },
+            {
+              Header: "File Name",
```
Comment (david-yz-liu) [inline app/assets/javascripts/Components/test_script_result_table.jsx]: Make sure you use the internationalized strings for the column headers, using `I18n.t(...)`.

File: app/assets/javascripts/Components/test_script_result_table.jsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import React from 'react';
+import {render} from 'react-dom';
+import ReactTable from 'react-table';
+import _ from 'lodash';
```
Comment (david-yz-liu) [inline app/assets/javascripts/Components/test_script_result_table.jsx]: `lodash` is not an existing dependency, nor should you be adding a new dependency. Look for another way to do what you want to do.

File: config/locales/es.yml [hunk 5]
```diff
@@ -1416,6 +1416,8 @@ es:
         hours:                    "horas"
       test_results_table:
         test_name:    "Nombre de la Prueba"
+        file_name: "UPDATE ME"
+        test_name: "UPDATE ME"
```
Comment (mishaschwartz) [inline config/locales/es.yml]: test_name is duplicated

File: config/locales/fr.yml [hunk 6]
```diff
@@ -1061,6 +1061,8 @@ fr:
         unlimited: "infinis"
       test_results_table:
         test_name: "Nom du Test"
+        file_name: "UPDATE ME"
+        test_name: "UPDATE ME"
```
Comment (mishaschwartz) [inline config/locales/fr.yml]: test_name is duplicated

File: config/locales/pt.yml [hunk 7]
```diff
@@ -1026,6 +1026,8 @@ Aplicar pênalti no final / Deduzir fichas de carência ( Deixando esta caixa de
         tokens_available: "<span class=\"prop_label\">Tokens disponíveis:</span> %{tokens}"
       test_results_table:
         test_name: "Nome do Teste"
+        file_name: "UPDATE ME"
+        test_name: "UPDATE ME"
```
Comment (mishaschwartz) [inline config/locales/pt.yml]: test_name is duplicated

**Comment by david-yz-liu** [comment]
Merged in squashed commit 1f59bff304f31da08cda7c865baf00f302b164cf.

**Pull request closed without merging**